oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -181,18 +183,19 @@ static ServiceAccountCredentials fromJson(
     String clientEmail = (String) json.get("client_email");
     String privateKeyPkcs8 = (String) json.get("private_key");
     String privateKeyId = (String) json.get("private_key_id");
+    String projectId = (String) json.get("project_id");
```

Comment:
`json.get()` returns `null` if the key is not present, and the rest of the code accepts null project ID values. Added a test case to verify this scenario.